src/main/java/Duke/UI/Ui.java

```diff
@@ -0,0 +1,126 @@
+package Duke.UI;
+
+import Duke.Parser.Parser;
+import Duke.Storage.Storage;
+import Duke.Tasks.Commands;
+import Duke.Tasks.Task;
+import Duke.Tasks.TaskList;
+
+import java.util.Optional;
+import java.util.Scanner;
+
+/**
+ * Controls any user interaction with the user.
+ * Examples: text output, error messages
+ */
+public class Ui {
+
+    private TaskList taskList;
+    private Storage storage;
+
+    private static final String SEPARATOR_LINE = "____________________________________________________________";
+
+    /**
+     * Constructor for an Ui handler.
+     *
+     * @param taskList
+     */
+    public Ui(TaskList taskList, Storage storage) {
+        this.taskList = taskList;
+        this.storage = storage;
+    }
+
+
+    /**
+     * Begins interaction with the user
+     */
+    public void beginLogging() {
+
+        printDivider();
+        String entranceMsg = "Hello! I'm Elon Musk.\n" +
+                "What can I do for you?";
+        System.out.println(entranceMsg);
+        printDivider();
+
+        String inputString = "";
+
+        Scanner keyboard = new Scanner(System.in);
+
+        loop:
+        while (true) {
+            try {
+                inputString = keyboard.nextLine();
+                printDivider();
+                boolean canContinue = Parser.parse(inputString, taskList, storage);
+                if (!canContinue) {
+                    break;
+                }
+                printDivider();
+            } catch (Exception e) {
```

Comment:
It might be better to avoid pokemon exceptions and catch specific ones for better debugging.